### PR TITLE
contracts cleanup

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
## Description

- Add `PancakeSwapFeature`.
- Remove TokenSpender/AllowanceTarget stuff from contracts, addresse, migrations.

### TODO
- [x] Regenerate wrappers
- [ ] Enable Pancake VIP in AS (will need to rebase off jacob's PR)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/zero-ex/contracts/src/features/PancakeSwapFeature.sol [hunk 1]
```diff
@@ -0,0 +1,401 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (dorothy-zbornak) [inline contracts/zero-ex/contracts/src/features/PancakeSwapFeature.sol:1]: Mostly minor changes here from `UniswapFeature`.
- Remove legacy allowance target logic.
- Remove greedy tokens logic.
- Fortunately the above two were appended logic and don't seem to impact the rest of the assembly by their removal.
- Update names, init hashes, addresses for the new AMMs/chain.

Comment (dorothy-zbornak) [reply contracts/zero-ex/contracts/src/features/PancakeSwapFeature.sol:1]: Here's a diff of the two files: https://gist.github.com/dorothy-zbornak/5a9d4e883ab1bac1f216354069781163

File: packages/contract-addresses/CHANGELOG.json [hunk 2]
```diff
@@ -1,4 +1,17 @@
 [
+    {
+        "version": "6.0.0-bsc.0",
```
Comment (dorothy-zbornak) [inline packages/contract-addresses/CHANGELOG.json:3]: Note: major bump

File: packages/migrations/CHANGELOG.json [hunk 3]
```diff
@@ -1,4 +1,12 @@
 [
+    {
+        "version": "8.0.0-bsc.0",
```
Comment (dorothy-zbornak) [inline packages/migrations/CHANGELOG.json:3]: Note: major bump

File: contracts/zero-ex/contracts/src/features/IPancakeSwapFeature.sol [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+
+
+/// @dev VIP uniswap fill functions.
+interface IPancakeSwapFeature {
+
+    /// @dev Efficiently sell directly to uniswap/sushiswap.
```
Comment (mzhu25) [inline contracts/zero-ex/contracts/src/features/IPancakeSwapFeature.sol:29]:
```suggestion
    /// @dev Efficiently sell directly to PancakeSwap/BakerySwap.
```

**Review by mzhu25 — APPROVED** [review]
me parece bien

File: contracts/zero-ex/contracts/src/features/TokenSpenderFeature.sol [hunk 5]
```diff
@@ -1,137 +0,0 @@
-// SPDX-License-Identifier: Apache-2.0
-/*
```
Comment (hysz) [inline contracts/zero-ex/contracts/src/features/TokenSpenderFeature.sol:2]: Is this still hooked up on mainnet?

File: contracts/zero-ex/contracts/src/features/UniswapFeature.sol [hunk 6]
```diff
@@ -39,12 +39,8 @@ contract UniswapFeature is
     string public constant override FEATURE_NAME = "UniswapFeature";
```
Comment (hysz) [inline contracts/zero-ex/contracts/src/features/UniswapFeature.sol:39]: Just wanted to confirm that integrators are no longer using the legacy proxy?

**Review by hysz — APPROVED** [review]
LGTM!

File: contracts/zero-ex/contracts/src/features/TokenSpenderFeature.sol (hunk 5 above)
Comment (dorothy-zbornak) [reply contracts/zero-ex/contracts/src/features/TokenSpenderFeature.sol:2]: yes

File: contracts/zero-ex/contracts/src/features/UniswapFeature.sol (hunk 6 above)
Comment (dorothy-zbornak) [reply contracts/zero-ex/contracts/src/features/UniswapFeature.sol:39]: API will stop supporting it march 31

**Pull request merged**